### PR TITLE
CLOUDSTACK-8856 Primary Storage Used(type tag with value 2) related t…

### PR DESCRIPTION
**Pull request by bvbharatk**
Primary Storage Used(type tag with value 2) related tag is not showing in listCapacity api response

### CODE REVIEW

**Comment by bvbharatk** [comment]
The above travis error is because of a failure to do git pull below are the relevant logs.


system_info
Build system information
Build language: java
Build image provisioning date and time
Wed Feb  4 18:22:50 UTC 2015
Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 12.04 LTS
Release:	12.04
Codename:	precise
Linux Version
2.6.32-042stab090.5
Cookbooks Version
23bb455 https://github.com/travis-ci/travis-cookbooks/tree/23bb455
GCC version
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
LLVM version
clang version 3.4 (tags/RELEASE_34/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
Pre-installed Ruby versions
ruby-1.9.3-p551
Pre-installed Node.js versions
v0.10.36
Pre-installed Go versions
1.4.1
Redis version
redis-server 2.8.19
riak version
2.0.2
MongoDB version
MongoDB 2.4.12
CouchDB version
couchdb 1.6.1
Neo4j version
1.9.4
Cassandra version
2.0.9
RabbitMQ Version
3.4.3
ElasticSearch version
1.4.0
Installed Sphinx versions
2.0.10
2.1.9
2.2.6
Default Sphinx version
2.2.6
Installed Firefox version
firefox 31.0esr
PhantomJS version
1.9.8
ant -version
Apache Ant(TM) version 1.8.2 compiled on December 3 2011
mvn -version
Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T17:29:23+00:00)
Maven home: /usr/local/maven
Java version: 1.7.0_76, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-oracle/jre
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "2.6.32-042stab090.5", arch: "amd64", family: "unix"
75.16s$ git clone --depth=50 https://github.com/apache/cloudstack.git apache/cloudstack
Cloning into 'apache/cloudstack'...
fatal: unable to access 'https://github.com/apache/cloudstack.git/': Couldn't resolve host 'github.com'
The command "eval git clone --depth=50 https://github.com/apache/cloudstack.git apache/cloudstack" failed. Retrying, 2 of 3.
Cloning into 'apache/cloudstack'...
fatal: unable to access 'https://github.com/apache/cloudstack.git/': Couldn't resolve host 'github.com'
The command "eval git clone --depth=50 https://github.com/apache/cloudstack.git apache/cloudstack" failed. Retrying, 3 of 3.
Cloning into 'apache/cloudstack'...
fatal: unable to access 'https://github.com/apache/cloudstack.git/': Couldn't resolve host 'github.com'
The command "eval git clone --depth=50 https://github.com/apache/cloudstack.git apache/cloudstack" failed 3 times.
The command "git clone --depth=50 https://github.com/apache/cloudstack.git apache/cloudstack" failed and exited with 128 during .
Your build has been stopped.

File: server/src/com/cloud/server/ManagementServerImpl.java [hunk 1]
```diff
@@ -2488,6 +2479,23 @@ public int compare(final SummedCapacity arg0, final SummedCapacity arg1) {
...
+                        if (capacity.getTotalCapacity() != 0) {
+                            capacity.setUsedPercentage((float)capacity.getUsedCapacity() / capacity.getTotalCapacity());
+                        } else {
+                            capacity.setUsedPercentage(0);
+                        }
+                        SummedCapacity summedCapacity = new SummedCapacity(capacity.getUsedCapacity(), capacity.getTotalCapacity(), capacity.getUsedPercentage(),
+                                capacity.getCapacityType(), capacity.getDataCenterId(), capacity.getPodId(), capacity.getClusterId());
+                        list.add(summedCapacity);
+                    }
                 }// End of for
```
Comment (pedro-martins) [inline server/src/com/cloud/server/ManagementServerImpl.java]: Hi @bvbharatk.
Could you extract lines 2489 - 2498 to a method like "configureUsageCapacities" and a test case to this new method? 
I believe you can also extract lines 2486 -2488 to a method like "createCapacities" with test cases too.
 
Another suggestion for you, I know that you don't touch in this part of the code, but how about you change the name of the variable declared at line 2464 to listOfSummedCapacity, the actual name "list" is a bit shallow.
 
If you find useful, a Java doc describing each methods task would be a plus.
Ty.

**Comment by rhtyd** [comment]
@bvbharatk please rebase against latest master and push -f, update on status of your PR

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 110
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by bvbharatk** [comment]
Rebased with 4.9

**Comment by bvbharatk** [comment]
Rebased with 4.9

File: server/src/com/cloud/server/ManagementServerImpl.java (hunk 1 above)
Comment (koteswararavipati) [inline server/src/com/cloud/server/ManagementServerImpl.java]: I test this issue manually and looks good.

Comment (koteswararavipati) [inline server/src/com/cloud/server/ManagementServerImpl.java]: I tested this issue manually and looks good.

**Comment by koteswararavipati** [comment]
LGTM:
I tested this issue manually and looks good.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 340
 Hypervisor xenserver
 NetworkType Advanced
 Passed=103
 Failed=1
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Comment by karuturi** [comment]
merging

**Pull request merged**